### PR TITLE
DOC: standardize capitalization of NEP headings

### PR DESCRIPTION
**Pull request by mdhaber**
More toward gh-16261.

gh-16261 reported that documentation headings currently have inconsistent capitalization schemes and should be standardized. gh-25016 fixed this for `rst` files except release notes and NEPS. This PR adjusts the NEPs.

The case of 'numpy', 'blas', 'scipy', 'python', 'lapack', 'fortran', 'api', 'pypi', 'cpu', 'abi', 'pr', 'github', 'ssh', 'git', 'f77', 'numba', 'f2py', 'cffi', 'bitgenerator', 'simd', 'swig', 'csv', 'ascii', 'json', 'dataframe', 'ieee', 'matlab', 'pyfort', ' c ', and 'nep' are unchanged. (Perhaps it would be a good idea to check the capitalization of some of these words throughout the documentation, but that is a different issue.)

<details>
Minimally modified version of script in gh-25016.

```python3
import os
import sys


def process(content):
    headings = set('#*=-^"')
    preserve = {'numpy', 'blas', 'scipy', 'python', 'lapack', 'fortran',
                'api', 'pypi', 'cpu', 'abi', 'pr', 'github', 'ssh', 'git',
                'f77', 'numba', 'f2py', 'cffi', 'bitgenerator', 'simd',
                'swig', 'csv', 'ascii', 'json', 'dataframe', 'ieee',
                'matlab', 'pyfort', ' c ', 'nep'}

    for i in range(1, len(content)):
        stripped = content[i].strip()
        x = set(stripped)
        if len(stripped) > 3 and not (x - headings):
            text = content[i-1]
            lower = text.lower()

            k = 0
            if text.startswith('NEP'):
                k = text.find(' — ') + 3

            sentence = lower[:k] + lower[k].upper() + lower[k+1:]

            for word in preserve:
                j = lower.find(word.lower())
                if j < 0:
                    continue
                word = text[j:j+len(word)]
                sentence = sentence[:j] + word + sentence[j + len(word):]

            content[i-1] = sentence

    return content


walk_dir = '/Users/matthaberland/Desktop/numpy'

print('walk_dir = ' + walk_dir)

# If your current working directory may change during script execution, it's
# recommended to immediately convert program arguments to an absolute path.
# Then the variable root below will be an absolute path as well. Example:
# walk_dir = os.path.abspath(walk_dir)
print('walk_dir (absolute) = ' + os.path.abspath(walk_dir))

skips = {''}

for root, subdirs, files in os.walk(walk_dir):
    print('--\nroot = ' + root)

    for filename in files:
        file_path = os.path.join(root, filename)

        # skips = {'vendored-meson', 'doc/source/release/', 'doc/neps/nep-'}
        skips = {'doc/neps/nep-'}

        skip = False
        for word in skips:
            if word in file_path:
                skip = True

        if not skip:
            continue

        print('\t- file %s (full path: %s)' % (filename, file_path))

        if file_path.endswith('.rst'):
            with open(file_path, 'r') as f:
                f_content = f.readlines()
                f_content = process(f_content)
            with open(file_path, 'w') as f:
                f.writelines(f_content)
                # list_file.write(('The file %s contains:\n' % filename).encode('utf-8'))
                # list_file.write(f_content)
                # list_file.write(b'\n')
```

</details>

### CODE REVIEW

File: doc/neps/nep-0012-missing-data.rst [hunk 1]
```diff
@@ -77,7 +77,7 @@ proposed elsewhere for customizing subclass ufunc behavior with a
 _numpy_ufunc_ member function would allow a subclass with a different
 default to be created.
 
-Unknown Yet Existing Data (NA)
+Unknown yet existing data (na)
```
Comment (mdhaber) [inline doc/neps/nep-0012-missing-data.rst:80]:
```suggestion
Unknown yet existing data (NA)
```

File: doc/neps/nep-0012-missing-data.rst [hunk 2]
```diff
@@ -98,7 +98,7 @@ such things to the theoretical limit is probably not worth it,
 and in many cases either raising an exception or returning all
 missing values may be preferred to doing precise calculations.
 
-Data That Doesn't Exist Or Is Being Skipped (IGNORE)
+Data that doesn't exist or is being skipped (ignore)
```
Comment (mdhaber) [inline doc/neps/nep-0012-missing-data.rst:101]:
```suggestion
Data that doesn't exist or is being skipped (IGNORE)
```

File: doc/neps/nep-0011-deferred-ufunc-evaluation.rst [hunk 3]
```diff
@@ -232,7 +232,7 @@ only when the error state is set to ignore all, but allow user control with
 use deferred evaluation, False would mean never use it, and None would
 mean use it only when safe (i.e. the error state is set to ignore all).
 
-Interaction With UPDATEIFCOPY
+Interaction with updateifcopy
```
Comment (mdhaber) [inline doc/neps/nep-0011-deferred-ufunc-evaluation.rst:235]:
```suggestion
Interaction with UPDATEIFCOPY
```

File: doc/neps/nep-0010-new-iterator-ufunc.rst [hunk 4]
```diff
@@ -1983,7 +1983,7 @@ functions.::
         ...:        composite_over_it(image1, image2))
     Out[18]: True
 
-Image Compositing With NumExpr
+Image compositing with numexpr
```
Comment (mdhaber) [inline doc/neps/nep-0010-new-iterator-ufunc.rst:1986]:
```suggestion
Image compositing with NumExpr
```

File: doc/neps/nep-0008-groupby_additions.rst [hunk 5]
```diff
@@ -1,7 +1,7 @@
 .. _NEP08:
 
 =============================================================
-NEP 8 —  A proposal for adding groupby functionality to NumPy
+NEP 8 —  a proposal for adding groupby functionality to NumPy
 =============================================================
```
Comment (mdhaber) [inline doc/neps/nep-0008-groupby_additions.rst:5]:
```suggestion
NEP 8 — a proposal for adding groupby functionality to NumPy
============================================================
```

File: doc/neps/nep-0008-groupby_additions.rst [hunk 6]
```diff
@@ -1,7 +1,7 @@
 .. _NEP08:
 
 =============================================================
```
Comment (mdhaber) [inline doc/neps/nep-0008-groupby_additions.rst:3]:
```suggestion
============================================================
```

File: doc/neps/nep-0007-datetime-proposal.rst [hunk 7]
```diff
@@ -619,7 +619,7 @@ list of tuples describing a data-format can itself be a tuple of
 Final considerations
 ====================
 
-Why the fractional time and events: [3Y/12]//50
+Why the fractional time and events: [3y/12]//50
```
Comment (mdhaber) [inline doc/neps/nep-0007-datetime-proposal.rst:622]:
```suggestion
Why the fractional time and events: [3Y/12]//50
```

File: doc/neps/nep-0007-datetime-proposal.rst [hunk 8]
```diff
@@ -505,7 +505,7 @@ With this, the above operation can be done as follows::
   Out[12]: array([366, 366, 366],  dtype="timedelta64[D]")
 
 
-dtype vs time units conversions
+Dtype vs time units conversions
```
Comment (mdhaber) [inline doc/neps/nep-0007-datetime-proposal.rst:508]:
```suggestion
dtype vs time units conversions
```

File: doc/neps/nep-0006-newbugtracker.rst [hunk 9]
```diff
@@ -1,7 +1,7 @@
 .. _NEP06:
 
 ===================================================
-NEP 6 — Replacing Trac with a different bug tracker
+NEP 6 — Replacing trac with a different bug tracker
```
Comment (mdhaber) [inline doc/neps/nep-0006-newbugtracker.rst:4]:
```suggestion
NEP 6 — Replacing Trac with a different bug tracker
```

File: doc/neps/nep-0004-datetime-proposal3.rst [hunk 10]
```diff
@@ -462,7 +462,7 @@ With this, the above operation can be done as follows::
   Out[12]: array([366, 366, 366],  dtype="timedelta64[D]")
 
 
-dtype vs time units conversions
+Dtype vs time units conversions
```
Comment (mdhaber) [inline doc/neps/nep-0004-datetime-proposal3.rst:465]:
```suggestion
dtype vs time units conversions
```

File: doc/neps/nep-0055-string_dtype.rst [hunk 11]
```diff
@@ -1,7 +1,7 @@
 .. _NEP55:
 
 =========================================================
-NEP 55 — Add a UTF-8 Variable-Width String DType to NumPy
+NEP 55 — Add a utf-8 variable-width string dtype to NumPy
```
Comment (mdhaber) [inline doc/neps/nep-0055-string_dtype.rst:4]: Usually dtype has no caps, but it's DType throughouht this NEP. What should be done?

```suggestion
NEP 55 — Add a UTF-8 variable-width string dtype to NumPy
```

File: doc/neps/nep-0040-legacy-datatype-impl.rst [hunk 12]
```diff
@@ -436,7 +436,7 @@ As such this step may move from the dispatching step (described above) to
 the implementation-specific code described below.
 
 
-DType-specific Implementation of the UFunc
+Dtype-specific implementation of the ufunc
```
Comment (mdhaber) [inline doc/neps/nep-0040-legacy-datatype-impl.rst:439]:
```suggestion
dtype-specific implementation of the ufunc
```

**Comment by mdhaber** [comment]
This should be pretty easy; I just need to know what to do about "UFunc" and "DType". 

I think it's pretty standard to make "dtype" all-lowercase, even at the beginning of the sentence. It's like "True" and "False" throughout the documentation - they are supposed to represent `True` and `False`, but the the backticks are omitted. (Personally, I'd add them, but not everyone agrees.)

It's a little tougher to figure out what to do with "ufunc". I've definitely seen it all-lowercase, and that's what I assumed in gh-25016, but it's debatable.

I'd prefer to make them lowercase always here, but then the headings might be inconsistent with the rest of the file.

**Comment by seberg** [comment]
In some places DType may be used to hint towards the Class rather than an instance `np.dtypes.Int64DType` is the class `np.dtype("i8")`.
You could argue that it should be in ticked when it does.  For UFunc, I doubt it matters much.  The discussion is probably the same (capitalized hints towards the API/class, while lower case would be more likely a specific ufunc).  But I think the distinction is not really confusing there.

**Comment by mdhaber** [comment]
Ok, I'll leave DType and dtype alone.

File: doc/neps/nep-0040-legacy-datatype-impl.rst [hunk 13]
```diff
@@ -317,7 +317,7 @@ This function is used multiple places and should probably be part of
 a redesigned casting API.
 
 
-DType handling in Universal functions
+dtype handling in universal functions
```
Comment (mdhaber) [inline doc/neps/nep-0040-legacy-datatype-impl.rst:320]:
```suggestion
DType handling in universal functions
```

File: doc/neps/nep-0040-legacy-datatype-impl.rst [hunk 14]
```diff
@@ -415,7 +415,7 @@ While option 2 may be less complex to reason about it remains to be seen
 whether it is sufficient for all (or most) use cases.
 
 
-Adjustment of Parametric output DTypes in UFuncs
+Adjustment of parametric output dtypes in ufuncs
```
Comment (mdhaber) [inline doc/neps/nep-0040-legacy-datatype-impl.rst:418]:
```suggestion
Adjustment of parametric output DTypes in UFuncs
```

File: doc/neps/nep-0040-legacy-datatype-impl.rst [hunk 15]
```diff
@@ -436,7 +436,7 @@ As such this step may move from the dispatching step (described above) to
 the implementation-specific code described below.
 
 
-DType-specific Implementation of the UFunc
+dtype-specific implementation of the ufunc
```
Comment (mdhaber) [inline doc/neps/nep-0040-legacy-datatype-impl.rst:439]:
```suggestion
DType-specific implementation of the UFunc
```

File: doc/neps/nep-0042-new-dtypes.rst [hunk 16]
```diff
@@ -1,7 +1,7 @@
 .. _NEP42:
 
 ==============================================================================
-NEP 42 — New and extensible DTypes
+NEP 42 — New and extensible dtypes
```
Comment (mdhaber) [inline doc/neps/nep-0042-new-dtypes.rst:4]:
```suggestion
NEP 42 — New and extensible DTypes
```

File: doc/neps/nep-0042-new-dtypes.rst [hunk 17]
```diff
@@ -262,7 +262,7 @@ dtype instances from DType classes.
 .. _nep42_DType class:
 
 ******************************************************************************
-The DType class
+The dtype class
```
Comment (mdhaber) [inline doc/neps/nep-0042-new-dtypes.rst:265]:
```suggestion
The DType class
```

File: doc/neps/nep-0042-new-dtypes.rst [hunk 18]
```diff
@@ -463,7 +463,7 @@ We review here the operations related to casting arrays:
 
 We show how casting arrays with ``astype(new_dtype)`` will be implemented.
 
-`Common DType` operations
+`Common dtype` operations
```
Comment (mdhaber) [inline doc/neps/nep-0042-new-dtypes.rst:466]:
```suggestion
`Common DType` operations
```

File: doc/neps/nep-0042-new-dtypes.rst [hunk 19]
```diff
@@ -984,7 +984,7 @@ important enough to be handled by NumPy using the presented methods.
     ``np.float128`` base class.
 
 
-DType discovery during array coercion
+dtype discovery during array coercion
```
Comment (mdhaber) [inline doc/neps/nep-0042-new-dtypes.rst:987]:
```suggestion
DType discovery during array coercion
```

File: doc/neps/nep-0042-new-dtypes.rst [hunk 20]
```diff
@@ -1214,7 +1214,7 @@ This is a general limitation, but round-tripping is always possible if
 Public C-API
 ******************************************************************************
 
-DType creation
+dtype creation
```
Comment (mdhaber) [inline doc/neps/nep-0042-new-dtypes.rst:1217]:
```suggestion
DType creation
```

File: doc/neps/nep-0043-extensible-ufuncs.rst [hunk 21]
```diff
@@ -1,7 +1,7 @@
 .. _NEP43:
 
 ==============================================================================
-NEP 43 — Enhancing the extensibility of UFuncs
+NEP 43 — Enhancing the extensibility of ufuncs
```
Comment (mdhaber) [inline doc/neps/nep-0043-extensible-ufuncs.rst:4]:
```suggestion
NEP 43 — Enhancing the extensibility of UFuncs
```

File: doc/neps/nep-0043-extensible-ufuncs.rst [hunk 22]
```diff
@@ -329,7 +329,7 @@ Where ``Integer`` is an abstract DType (compare NEP 42).
 .. _steps_of_a_ufunc_call:
 
 ****************************************************************************
-Steps involved in a UFunc call
+Steps involved in a ufunc call
```
Comment (mdhaber) [inline doc/neps/nep-0043-extensible-ufuncs.rst:332]:
```suggestion
Steps involved in a UFunc call
```

File: doc/neps/nep-0055-string_dtype.rst [hunk 23]
```diff
@@ -1,7 +1,7 @@
 .. _NEP55:
 
 =========================================================
-NEP 55 — Add a UTF-8 Variable-Width String DType to NumPy
+NEP 55 — Add a UTF-8 variable-width string dtype to NumPy
```
Comment (mdhaber) [inline doc/neps/nep-0055-string_dtype.rst:4]:
```suggestion
NEP 55 — Add a UTF-8 variable-width string DType to NumPy
```

File: doc/neps/nep-0041-improved-dtype-support.rst [hunk 24]
```diff
@@ -737,7 +737,7 @@ a later step in the implementation to reduce the complexity of the initial
 implementation.
 
 
-C-API Changes to the UFunc Machinery (4)
+C-API changes to the ufunc machinery (4)
```
Comment (mdhaber) [inline doc/neps/nep-0041-improved-dtype-support.rst:740]:
```suggestion
C-API changes to the UFunc machinery (4)
```

File: doc/neps/nep-0010-new-iterator-ufunc.rst [hunk 25]
```diff
@@ -1,7 +1,7 @@
 .. _NEP10:
 
 ==============================================
-NEP 10 — Optimizing Iterator/UFunc performance
+NEP 10 — Optimizing iterator/ufunc performance
```
Comment (mdhaber) [inline doc/neps/nep-0010-new-iterator-ufunc.rst:4]:
```suggestion
NEP 10 — Optimizing iterator/UFunc performance
```

File: doc/neps/nep-0010-new-iterator-ufunc.rst [hunk 26]
```diff
@@ -1680,7 +1680,7 @@ references, and add ``NPY_ITER_WRITEABLE_REFERENCES`` to the flags:
         return ret;
     }
 
-Python Lambda UFunc Example
+Python lambda ufunc example
```
Comment (mdhaber) [inline doc/neps/nep-0010-new-iterator-ufunc.rst:1683]:
```suggestion
Python lambda UFunc example
```

File: doc/neps/nep-0011-deferred-ufunc-evaluation.rst [hunk 27]
```diff
@@ -1,7 +1,7 @@
 .. _NEP11:
 
 ==================================
-NEP 11 — Deferred UFunc evaluation
+NEP 11 — Deferred ufunc evaluation
```
Comment (mdhaber) [inline doc/neps/nep-0011-deferred-ufunc-evaluation.rst:4]:
```suggestion
NEP 11 — Deferred UFunc evaluation
```

File: doc/neps/nep-0012-missing-data.rst [hunk 28]
```diff
@@ -455,7 +455,7 @@ New functions added to the ndarray are::
         >>> a.flags.maskna = True
         >>> a.flags.ownmaskna = True
 
-Element-wise UFuncs With Missing Values
+Element-wise ufuncs with missing values
```
Comment (mdhaber) [inline doc/neps/nep-0012-missing-data.rst:458]: Everywhere in the paragraph, ufunc is not capitalized, so this follows suit.

File: doc/neps/nep-0013-ufunc-overrides.rst [hunk 29]
```diff
@@ -415,7 +415,7 @@ with them in ufuncs.
 
 .. _neps.ufunc-overrides.turning-ufuncs-off:
 
-Turning Ufuncs off
+Turning ufuncs off
```
Comment (mdhaber) [inline doc/neps/nep-0013-ufunc-overrides.rst:418]:
```suggestion
Turning Ufuncs off
```

File: doc/neps/nep-0013-ufunc-overrides.rst [hunk 30]
```diff
@@ -622,7 +622,7 @@ or NumPy's dispatch rules, but not some mixture of both at the same time.
 
 .. _neps.ufunc-overrides.list-of-operators:
 
-List of operators and NumPy Ufuncs
+List of operators and NumPy ufuncs
```
Comment (mdhaber) [inline doc/neps/nep-0013-ufunc-overrides.rst:625]:
```suggestion
List of operators and NumPy Ufuncs
```

File: doc/neps/nep-0013-ufunc-overrides.rst [hunk 31]
```diff
@@ -1,7 +1,7 @@
 .. _NEP13:
 
 ==========================================
-NEP 13 — A mechanism for overriding Ufuncs
+NEP 13 — A mechanism for overriding ufuncs
```
Comment (mdhaber) [inline doc/neps/nep-0013-ufunc-overrides.rst:4]:
```suggestion
NEP 13 — A mechanism for overriding Ufuncs
```

**Comment by mdhaber** [comment]
OK @seberg I think this one is ready for a pass, if you're willing.

File: doc/neps/nep-0008-groupby_additions.rst [hunk 32]
```diff
@@ -1,8 +1,8 @@
 .. _NEP08:
 
-=============================================================
-NEP 8 —  A proposal for adding groupby functionality to NumPy
-=============================================================
+============================================================
+NEP 8 — a proposal for adding groupby functionality to NumPy
```
Comment (seberg) [inline doc/neps/nep-0008-groupby_additions.rst:4]:
```suggestion
NEP 8 — A proposal for adding groupby functionality to NumPy
```
You didn't change the others, and this is just as well.

**Review by seberg — APPROVED** [review]
Looks fine to me, one nitpick.

File: doc/neps/nep-0008-groupby_additions.rst (hunk 32 above)
Comment (mdhaber) [reply doc/neps/nep-0008-groupby_additions.rst:4]: Thanks. I missed this when going back and fixing false positives.

**Pull request merged**